src/main/resources/avro/alleleAnnotations.avdl

```diff
@@ -0,0 +1,198 @@
+VariantSet to a specific set of reference data using specific software tools.
+A VariantAnnotationSet contains information on reference data and software
+versions used in calculating the annotation; it is essential this information
+is exhaustive.
+
+*/
+
+/**
+An AnalysisResult record holds the output of a prediction package such
+as SIFT on a specific allele.
+*/
+record AnalysisResult {
+
+  /** The ID of the analysis record for this result */
+  string analysisId;
+
+  /** The text-based result for this analysis */
+  union { null, string} result;
+
+  /** The numeric score for this analysis  */
+  union { null, int} score;
+
+}
+
+/**
+An allele location record holds the location of an allele relative to a
+non-genomic coordinate system such as a CDS or protein and holds the
+reference and alternate sequence where appropriate
+*/
+record AlleleLocation {
+
+  /** Relative start position of the allele in this coordinate system */
+  int start ;
+
+  /** Relative end position of the allele in this coordinate system  */
+  union { null, int } end = null;
+
+  /** Reference sequence in feature (this should be the codon at CDS level) */
+  union { null, string } referenceSequence = null;
+
+  /** Alternate sequence in feature (this should be the codon at CDS level) */
+  union { null, string } alternateSequence = null;
+}
+
+/**
+Impact is a simple prioritization for the effect of an allele which is used
+in the annotation record.
+IMPORTANT:
+ Prioritization methods are a crude estimates and are not assumed to be
+ reliable: a 'HIGH' Impact may actually not cause any disruption
+ in protein function or expression.
+*/
+enum Impact {
+ HIGH,      // Variant highly disrupts protein function
+ MODERATE,  // Moderately disrupts protein function
+ LOW,       // Low disruption of protein impact
+ MODIFIER   // No known effect
+}
+
+/**
+A VariantAnnotationSet record groups VariantAnnotation records. It is derived
+from a VariantSet and holds information describing the software and reference
+data used in the annotation.
+*/
+record VariantAnnotationSet {
+
+  /** The ID of the variant annotation set record */
+  string id;
+
+  /** The ID of the variant set to which this annotation set belongs */
+  string variantSetId;
+
+  /** The variant annotation set name. */
+  union { null, string } name = null;
+
+  /**
+  Analysis details. It is essential to supply versions for all software and
+  reference data used.
+  */
+  Analysis analysis;
+}
+
+/**
+A HGVSAnnotation record holds Human Genome Variation Society descriptions
+of the sequence change with respect to genomic, transcript and protein
+sequences. See: http://www.hgvs.org/mutnomen/recs.html.
+Descriptions should be provided at genomic level. Descriptions at transcript
+level should be provided when the allele lies within a transcript. Descriptions
+at protein level should be provided when the allele lies within the translated
+sequence or stop codon.
+*/
+record HGVSAnnotation {
```

Comment:
GA4GH is intended to be providing an API that is data-set agnostic, so
both RefSeq and Ensembl should be equal valid to include in a data
set.

Of course, being able to resolve annotations against multiple
versions of dataset would be ideal.

Reece Hart <notifications@github.com> writes:
> In src/main/resources/avro/alleleAnnotations.avdl:
> 
> > +  Analysis details. It is essential to supply versions for all software and
> > +  reference data used.
> > +  */
> > +  Analysis analysis;
> > +}
> > +
> > +/**
> > +A HGVSAnnotation record holds Human Genome Variation Society descriptions
> > +of the sequence change with respect to genomic, transcript and protein
> > +sequences. See: http://www.hgvs.org/mutnomen/recs.html.
> > +Descriptions should be provided at genomic level. Descriptions at transcript
> > +level should be provided when the allele lies within a transcript. Descriptions
> > +at protein level should be provided when the allele lies within the translated
> > +sequence or stop codon.
> > +*/
> > +record HGVSAnnotation {
> 
> Do we have a sense of how these will be used and whether they will be searched?
> 
> For example, how do we handle ENST v. RefSeq transcripts? One answer might be
> that we're agnostic, but then searching will be hampered which kinda defeats a
> core tenet of GA4GH.
> 
> In other words, is the representation here unambiguous, or might it reasonably
> depend on an end-user preference? If the latter, it might be better to let
> end-user software deal with the transformations.
> 
> —
> Reply to this email directly or view it on GitHub.*
>